src/Microsoft.Data.Sqlite.Core/SqliteCommand.cs

```diff
@@ -82,13 +85,36 @@ public override CommandType CommandType
         /// Gets or sets the SQL to execute against the database.
         /// </summary>
         /// <value>The SQL to execute against the database.</value>
-        public override string CommandText { get; set; }
+        public override string CommandText
+        {
+            get { return _commandText; }
+            set
+            {
+                if (!value.Equals(_commandText))
+                {
+                    DisposePreparedStatements();
+                    _commandText = value;
+                }
+            }
+        }
 
         /// <summary>
         /// Gets or sets the connection used by the command.
         /// </summary>
         /// <value>The connection used by the command.</value>
-        public new virtual SqliteConnection Connection { get; set; }
+        public new virtual SqliteConnection Connection
+        {
+            get { return _connection; }
+            set
+            {
+                if (value != _connection)
+                {
+                    DisposePreparedStatements();
+                    _connection = value;
```

Comment:
In order to find it in the old collection I now added a new field with the weak reference to the SqliteCommand. I'm not sure if this is the best solution.